Source/WebGPU/WebGPU/Sampler.mm

```diff
@@ -31,6 +31,11 @@
 #import <cmath>
 #import <wtf/TZoneMallocInlines.h>
 
+// FIXME: remove once FB6018791 is addressed in Foundation
+@interface NSObject (NSPointerArray_15396578)
```

Comment:
Mike Ash suggested NSPointerArray instead of overriding dealloc as originally done in https://commits.webkit.org/283645@main

However, there is an outstanding bug tracked by FB6018791 / radar://15396578 so with some reflection it can be concluded that `_markNeedsCompaction` needs to be called prior to calling `compact` for `compact` to work as intended.